mobile/native/src/channel_fee.rs

```diff
@@ -51,24 +55,38 @@ impl Subscriber for ChannelFeePaymentSubscriber {
 }
 
 impl ChannelFeePaymentSubscriber {
-    pub fn new() -> Self {
+    pub fn new(channel_manager: Arc<ChannelManager>) -> Self {
         Self {
             open_channel_tx: Arc::new(Mutex::new(None)),
+            channel_manager,
         }
     }
 
     /// Attempts to pay the transaction fees for opening an inbound channel.
     fn pay_funding_transaction_fees(&self, amount_msats: u64) -> Result<()> {
-        let transaction = match self.get_funding_transaction() {
-            Some(transaction) => transaction,
+        let (channel_id, transaction) = match self.get_funding_transaction() {
+            Some((channel_id, transaction)) => (channel_id, transaction),
             None => {
                 tracing::debug!("No pending funding transaction found!");
                 return Ok(());
             }
         };
 
-        tracing::debug!("Trying to pay channel opening fees of {}", transaction.fee);
         let funding_tx_fees_msats = (transaction.fee * 1000) as u64;
+
+        tokio::task::block_in_place(|| {
+            tracing::debug!(
+                "Waiting for outbound capacity on channel {} to pay {funding_tx_fees_msats} msats jit channel opening fee.",
+                hex::encode(channel_id)
+            );
+            Handle::current()
+                .block_on(self.wait_for_outbound_capacity(channel_id, funding_tx_fees_msats))
+        })?;
```

Comment:
👍 I think this is okay. You could write a non-async `wait_for_outbound_capacity` so that you don't need to do all this, but I guess `tokio::time::timeout` is very convenient.